### ISSUE TITLE
Support for Service Bus Premium Customer-Managed Key Encryption

### ISSUE THREAD

**Issue by di669** [hashicorp/terraform-provider-azurerm#7116]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Azure Service Bus Premium provides users with the option to encryption data with customer managed keys.  In order to do this, an SP must be assigned to the Service Bus, the Service Bus SP  must be granted access to the Key Vault and then we assign the customer key to be used. Changes that will have to be implemented are:

- Creating an SP in the Service Bus 

- Creating a resource to assign customer key to the Service Bus.

### New or Affected Resource(s)

* azurerm_servicebus_namespace
* azurerm_servicebus_customer_managed_key

### Potential Terraform Configuration

```hcl
data "azurerm_key_vault" "vault" {
  name                = var.vault_name
  resource_group_name = var.vault_resource_group
}

resource "azurerm_servicebus_namespace" "we-service-bus_primary_namespace" {
  name                = "primary namespace"
  location            = var.we_location
  resource_group_name = var.we_resource_group_name
  sku                 = var.service_bus_sku
  capacity            = var.service_bus_sku == "premium"? var.service_bus_capacity: 0 // when sku is basic or standard capacity must be 0
  zone_redundant = true

#allow for SP creation
  identity {
    type = "SystemAssigned"
  }
}

#Assign Service Bus SP Permission to Key-Vault
resource "azurerm_key_vault_access_policy" "sb-we" {
  key_vault_id = data.azurerm_key_vault.vault.id
  tenant_id    = var.tenant_id
  object_id    = azurerm_servicebus_namespace.we-service-bus_primary_namespace.identity.0.principal_id

  key_permissions    = ["get", "create", "list", "restore", "recover", "unwrapkey", "wrapkey", "purge", "encrypt", "decrypt", "sign", "verify"]
  secret_permissions = ["get"]
}

#New resource to add customer managed key to Service Bus
resource "azurerm_servicebus_customer_managed_key" "managed_key" {
  storage_account_id = azurerm_servicebus_namespace.we-service-bus_primary_namespace.id
  key_vault_id       = data.azurerm_key_vault.vault.id
  key_name           = data.azurerm_key_vault_key.vault-key.name
  key_version        = data.azurerm_key_vault_key.vault-key.version
}
```

### References

* #0000

**Comment by spacexii** [comment]
Is there an update?

**Comment by sach83** [comment]
is there any update on it?

**Comment by shanmugarajeswaran** [comment]
Can we have any update on this please

**Comment by catriona-m** [comment]
Thanks for opening this @di669. It looks like this has now been resolved by #15601, so this can now be closed. Please feel free to reopen if this is not the case. Thanks!
